packages/mds-db/index.ts

```diff
@@ -1513,16 +1522,19 @@ async function readEventsRangeExclusive(
   }
   const where = conditions.length > 0 ? `WHERE ${conditions.join(' AND ')}` : ''
   const { rows } = await exec(
-    `SELECT E.*, T.lat, T.lng FROM ${schema.EVENTS_TABLE} E JOIN ${schema.TELEMETRY_TABLE} T ON E.device_id = T.device_id AND E.telemetry_timestamp = T.timestamp ${where} ORDER BY E.timestamp, E.device_id LIMIT ${take}`,
+    `SELECT E.*, T.lat, T.lng FROM ${schema.EVENTS_TABLE} E LEFT JOIN ${schema.TELEMETRY_TABLE} T ON E.device_id = T.device_id AND E.telemetry_timestamp = T.timestamp ${where} ORDER BY E.timestamp, E.device_id LIMIT ${take}`,
     vals.values()
   )
   return rows.map(({ lat, lng, telemetry_timestamp, ...event }) => ({
     ...event,
     telemetry_timestamp,
-    telemetry: {
-      timestamp: telemetry_timestamp,
-      gps: { lat, lng }
-    }
+    telemetry:
+      lat !== null && lng !== null
```

Comment:
Could just be: `telemetry: lat && lng ? { timestamp: telemetry_timestamp, gps: { lat, lng } } : null`

Reply:
I guess that's safe ... as long as you want to treat `{ lat: 0, lng: 0 }` as `null` also.

Reply:
or undefined :)

Reply:
Coming from db it will never be undefined

Reply:
doh